tests/rptest/services/cloud_broker.py

```diff
@@ -16,6 +25,10 @@ def __init__(self, pod, kubectl, logger) -> None:
         self._meta = pod['metadata']
         self.name = self._meta['name']
 
+        # Backward compatibility
+        # Various classes will use this to hash and compare nodes
+        self.account(pod)
```

Comment:
Why does this `account` need to be a class, that we instantiate?

Actually I don't really understand what is happening here at all :).

After this line, the CloudBroker object is unchanged right?

Looking further down, may be the intent is something like `self.account = account(pod)`?

Reply:
No this is a mimic of ducktape mechanism that is defined [here](https://github.com/redpanda-data/ducktape/blob/fba40dc3282394d29dc3002bfb1b448632b6d3db/ducktape/cluster/remoteaccount.py#L376).

Ducktape uses separate class to define sophisticated way to do SSH to node and run commands: `Linux_RemoteAccount`. Since we need to be consistent with normal logs extraction when searching for errors inside them in `raise_on_badlogs`, we just defining fake class with one needed attribute: `hostname` which is consumed in multiple places as if it is duck-tape's node.account.<RemoteAccount> class

Reply:
Primary cause for this dance is [here](https://github.com/redpanda-data/redpanda/pull/17772/files/97928f508243a02b967024d9b5d01044bb5844b8#diff-2a5b54ab60b67fd90a8271177be0b17ad3601e1ae2cadf522a1776568f782929R57). 

Class BafLogLines used in separate module and there was a discussion couple of weeks back not to overcomplicate it with inheritance (which was original solution)